engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/MigrateProcessInstanceTest.java

```diff
@@ -313,6 +313,63 @@ public void shouldWriteElementMigratedEventForServiceTaskWithNewId() {
         .hasElementId("B");
   }
 
+  @Test
+  public void shouldWriteElementMigratedEventForUserTaskWithJobWorkerImplementation() {
+    // given
+    final String processId = helper.getBpmnProcessId();
+    final String targetProcessId = helper.getBpmnProcessId() + "2";
+
+    final var deployment =
+        ENGINE
+            .deployment()
+            .withXmlResource(
+                Bpmn.createExecutableProcess(processId)
+                    .startEvent()
+                    .userTask("A")
+                    .endEvent()
+                    .done())
+            .withXmlResource(
+                Bpmn.createExecutableProcess(targetProcessId)
+                    .startEvent()
+                    .userTask("A")
```

Comment:
🔧 Let's use a different element id for the target element. Previously, we had issue with migrating elements with different id and it was not caught in tests. I know, in this case, it shouldn't be a problem but it would be good if we make it a common practice. Alternatively, a new test case can be introduced.